src/lsstseries/ensemble.py

```diff
@@ -284,6 +284,66 @@ def prune(self, threshold=50, col_name=None):
 
         return self
 
+    def bin_sources(self, time_window=1.0, additional_cols=None, use_map=True, **kwargs):
+        """Bin sources on within a given time range to improve the estimates.
+
+        Notes
+        -----
+        * This should only be used for slowly varying sources where we can
+        treat the source as constant within `time_window`.
+
+        * As a default the function only aggregates and keeps the id, band,
+        time, flux, and flux error columns. Additional columns can be preserved
+        by providing the mapping of column name to aggregation function with the
+        `additional_cols` parameter.
+
+        Parameters
+        ----------
+        time_window : `float` (optional)
+            The time range (in days) over which to consider observations in the same bin.
+        additional_cols : `dict` (optional)
+            A dictionary mapping column name to aggregation method.
+            Example: {"my_value_1": "mean", "my_value_2": "max"}
+        use_map : `boolean` (optional)
+            Determines whether `dask.dataframe.DataFrame.map_partitions` is
+            used (True). Using map_partitions is generally more efficient, but
+            requires the data from each lightcurve is housed in a single
+            partition. If False, a groupby will be performed instead.
+        """
+        # Bin the time and add it as a column.
+        tmp_time_col = "tmp_time_for_aggregation"
+        if tmp_time_col in self._source.columns:
+            raise KeyError(f"Column '{tmp_time_col}' already exists in source table.")
+        self._source[tmp_time_col] = self._source[self._time_col].apply(
+            lambda x: np.floor(x / time_window) * time_window, meta=pd.Series(dtype=float)
+        )
+
+        # Set up the aggregation functions for the time and flux columns.
+        aggr_funs = {self._time_col: "mean", self._flux_col: "mean"}
```

Comment:
Maybe more of a @nevencaplar question, is mean all we would need to support? When I think of binning I tend to think of a sum.

It also looks like the code is fairly rigid for this, since the error calculation depends on the aggregation done on flux. So I'm not neccesarily advocating for it to be generalized here, but was just curious if mean is best serving.

Reply:
Good question. The way the code is structured, you can actually overwrite the aggregation functions for time, flux, and error by including them in the additional columns dictionary (since those overwrite the items in aggr_funs). I hadn't thought of this as a use case, so I didn't include it in the docstring.